_posts/2018-04-15-lazy-loading-images.md

```diff
@@ -0,0 +1,207 @@
+![Image Optim](https://res.cloudinary.com/dsteinel/image/upload/v1524256663/imageoptim.png)
+
+
+### Lazy loading images
+Lazy loading means, that you only load the images which are actually in your viewport. So if the user lands on your website they don’t need to load the images which are at the very bottom of your page. So what we want to do is only show the user the image which is in the viewport.
+
+First we need some images in the DOM, so let’s create some dummy content.
+Make sure you apply some styling so that you have some scrolling on the page and you do not show all images above the fold (in the current viewport).
+
+{% highlight html %}
+<div class="image-container">
+  <ul class="image__list">
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796300/pexels-photo-559768.jpg"
+        alt="blue lake with surrounding mountains">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796305/pexels-photo-104336.jpg"
+        alt="street in the middle of a forrest">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796376/les-anderson-175603.jpg"
+        alt="two elder people opening togehter a christmas present">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796381/pexels-photo-737108.jpg"
+        alt="black and white clothes hanging on a clothes rail">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796475/tobias-van-schneider-122288.jpg"
+        alt="book collection photographed from top">
+    </li>
+    <li class="image__list-item">
+      <img class="image__list-image image__list-lazy-image" data-src="http://res.cloudinary.com/dsteinel/image/upload/c_scale,w_450/v1523796300/pexels-photo-559768.jpg"
+        alt="brown meadow with a tower in the very back">
+    </li>
+  </ul>
+</div>
+{% endhighlight %}
+
+{% highlight css %}
+.image-container {
+  display: block;
+  margin: 0 auto;
+  width: 450px;
+}
+
+.image__list-lazy-image { 
+  display: none;
+}
+
+.image__list-image {
+  display: block;
+  background-color: #333;
+}
+
+.image__list {
+  list-style-type: none;
+}
+
+.image__list-item {
+  width: 100%;
+  height: 300px;
+  margin-bottom: 40px;
+}
+{% endhighlight %}
```

Comment:
Does the code example really need to be that complex? I think you can make your point with less markup/ CSS.

Reply:
Good point!